eva-server/src/main/java/uk/ac/ebi/eva/server/ws/RegionWSServer.java

```diff
@@ -55,98 +64,70 @@ public RegionWSServer() {
         }
 
-        if (histogram) {
-            if (regions.size() != 1) {
-                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                return setQueryResponse("Sorry, histogram functionality only works with a single region");
-            } else {
-                if (interval > 0) {
-                    queryOptions.put("interval", interval);
-                }
-                return setQueryResponse(
-                        variantMongoDbAdaptor.getVariantFrequencyByRegion(regions.get(0), queryOptions));
-
-            }
-        } else if (regionsSize <= 1000000) {
-            if (regions.isEmpty()) {
-                if (!queryOptions.containsKey("id") && !queryOptions.containsKey("gene")) {
-                    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                    return setQueryResponse("Some positional filer is needed, like region, gene or id.");
-                } else {
-                    return setQueryResponse(variantMongoDbAdaptor.getAllVariants(queryOptions));
-                }
-            } else {
-                return setQueryResponse(variantMongoDbAdaptor.getAllVariantsByRegionList(regions, queryOptions));
-            }
+        List<Region> regions = Region.parseRegions(regionId);
+        List<VariantEntity> variantEntities;
+
+        PageRequest pageRequest = Utils.getPageRequest(queryOptions);
+
+        if (regions.size() > 1) {
+            variantEntities =
+                    variantEntityRepository.findByRegionsAndComplexFilters(regions, studies, consequenceType,
+                                                                           mafOperator, mafvalue, polyphenScoreOperator,
+                                                                           polyphenScoreValue, siftScoreOperator,
+                                                                           siftScoreValue, pageRequest);
+        } else if (regions.size() == 1) {
+            Region region = regions.get(0);
+            variantEntities =
+                    variantEntityRepository.findByRegionAndComplexFilters(region.getChromosome(), region.getStart(),
+                                                                          region.getEnd(), studies, consequenceType,
+                                                                          mafOperator, mafvalue, polyphenScoreOperator,
+                                                                          polyphenScoreValue, siftScoreOperator,
+                                                                          siftScoreValue, pageRequest);
         } else {
-            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-            return setQueryResponse("The total size of all regions provided can't exceed 1 million positions. "
-                    + "If you want to browse a larger number of positions, please provide the parameter 'histogram=true'");
+            throw new IllegalArgumentException();
```

Comment:
So, are we following the throw exceptions that are translated or the setQueryResponse? @cyenyxe